earth_enterprise/src/common/SharedString.h

```diff
@@ -42,6 +43,7 @@ class SharedString {
         std::unordered_map<uint32_t, std::string> refFromKeyTable;
         std::unordered_map<std::string, uint32_t> keyFromRefTable;
         static uint32_t nextID;
+        mutable std::mutex m;
```

Comment:
We may want to consider using shared_mutex and shared_lock/unique_lock where appropriate. That would allow different threads to call `size` and `RefFromKey` simultaneously. With the sheer number of calls that will be made to `RefFromKey`, I think this might be worthwhile.
Caveat: I actually haven't used it before, so I'm speaking hypothetically here, but it might be good.

Reply:
@tst-jkent pointed out that shared_mutex is only available in C++17.